### PR TITLE
Adding categorised release notes for the v1.4 release

### PR DESCRIPTION
**Pull request by aburdenthehand**
Release notes for our upcoming v1.4 release.

Source: https://github.com/kubevirt/kubevirt/releases/tag/v1.4.0-rc.0
 
 To be merged on 12 Nov

### CODE REVIEW

File: docs/release_notes.md [hunk 1]
```diff
@@ -5,6 +5,147 @@ hide:
 
 # KubeVirt release notes
 
+## v1.4.0
+
+Release on: Wed Nov 12 2024
```
Comment (EdDev) [inline docs/release_notes.md:10]: 13

File: docs/release_notes.md [hunk 2]
```diff
@@ -5,6 +5,147 @@ hide:
...
+- [[PR #12194]](https://github.com/kubevirt/kubevirt/pull/12194) [mhenriks] VM supports kubevirt.io/immediate-data-volume-creation: "false" which delays creating DataVolumeTemplates until VM is started
+- [[PR #12254]](https://github.com/kubevirt/kubevirt/pull/12254) [jkinred] * Reduced the severity of log messages when a `VolumeSnapshotClass` is not found. When snapshots are not enabled for a volume, the reason will still be displayed in the `status.volumeSnapshotStatuses` field of a `VirtualMachine` resource.
+- [[PR #12601]](https://github.com/kubevirt/kubevirt/pull/12601) [mhenriks] vmsnapshot: Enable status subresource for snapshot.kubevirt.io api group
+
+### SIG-network
+- [[PR #13024]](https://github.com/kubevirt/kubevirt/pull/13024) [EdDev] network binding plugin: Introduce a new `managedTap` `domainAttachmentType`
+- [[PR #13060]](https://github.com/kubevirt/kubevirt/pull/13060) [EdDev] Network binding plugins feature is declared as Beta.
+- [[PR #12613]](https://github.com/kubevirt/kubevirt/pull/12613) [orelmisan] Bridge binding: Static routes to subnets containing the pod's NIC IP address are passed to the VM.
```
Comment (EdDev) [inline docs/release_notes.md:101]: This is a bug fix related to networking.
I do not know if it needs to be left here, moved to the bug-fix section or both (like the API change ones).

Comment (aburdenthehand) [reply docs/release_notes.md:101]: Thanks @EdDev - I'll move it to bugfixes

**Review by EdDev — APPROVED** [review]

**Pull request merged**